File: Customer/src/main/java/com/postman/alt/service/AttachmentService.java
package com.postman.alt.service;

import com.postman.alt.service.dto.AttachmentCreateRequest;
import com.postman.alt.service.dto.AttachmentResponse;

import java.util.List;
import java.util.UUID;

public interface AttachmentService {

    List<AttachmentResponse> list(UUID workItemId, UUID requesterId);

    AttachmentResponse create(UUID workItemId, UUID requesterId, AttachmentCreateRequest request);

    void delete(UUID attachmentId, UUID requesterId);
}

File: client/src/app/core/services/attachment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../config/api.config';
import { AttachmentCreateRequest, AttachmentResponse } from '../models/attachment.model';

const BASE = `${API_BASE_URL}/api`;

@Injectable({ providedIn: 'root' })
export class AttachmentService {
  private readonly http = inject(HttpClient);

  list(workItemId: string): Observable<AttachmentResponse[]> {
    return this.http.get<AttachmentResponse[]>(`${BASE}/work-items/${workItemId}/attachments`);
  }

  create(workItemId: string, request: AttachmentCreateRequest): Observable<AttachmentResponse> {
    return this.http.post<AttachmentResponse>(`${BASE}/work-items/${workItemId}/attachments`, request);
  }

  delete(attachmentId: string): Observable<void> {
    return this.http.delete<void>(`${BASE}/attachments/${attachmentId}`);
  }
}
